File: templates/csharp/app-mvvm/ViewModels/MainViewModel.cs
#if (CommunityToolkitChosen)
using CommunityToolkit.Mvvm.ComponentModel;
#elif (ReactiveUIToolkitChosen)
using ReactiveUI;
#endif

namespace AvaloniaAppTemplate.ViewModels;

#if (CommunityToolkitChosen)
public partial class MainViewModel : ViewModelBase
{
    [ObservableProperty]
#if (UsePartialProperties)
    public partial string Greeting { get; set; } = "Welcome to Avalonia!";
#else
    private string _greeting = "Welcome to Avalonia!";
#endif
}
#elif (ReactiveUIToolkitChosen)
public class MainViewModel : ViewModelBase
{
    private string _greeting = "Welcome to Avalonia!";

    public string Greeting
    {
        get => _greeting;
        set => this.RaiseAndSetIfChanged(ref _greeting, value);
    }
}
#endif

File: templates/fsharp/app-mvvm/ViewModels/MainViewModel.fs
namespace AvaloniaAppTemplate.ViewModels

#if (ReactiveUIToolkitChosen)
open ReactiveUI
#endif

type MainViewModel() =
    inherit ViewModelBase()

    let mutable greeting = "Welcome to Avalonia!"

    member this.Greeting
        with get () = greeting
#if (CommunityToolkitChosen)
        and set value = this.SetProperty(&greeting, value) |> ignore
#elif (ReactiveUIToolkitChosen)
        and set value = this.RaiseAndSetIfChanged(&greeting, value) |> ignore
#endif
